### PR TITLE
Replace constant time from ring with subtle

### PR DESCRIPTION
**Pull request by ctron**
This PR tries to start working towards #58 … but chopping it down into smaller pieces.

### CODE REVIEW

File: src/verification.rs [hunk 1]
```diff
@@ -480,9 +479,8 @@ pub trait NonceVerifier {
 impl NonceVerifier for &Nonce {
     fn verify(self, nonce: Option<&Nonce>) -> Result<(), String> {
         if let Some(claims_nonce) = nonce {
-            if verify_slices_are_equal(claims_nonce.secret().as_bytes(), self.secret().as_bytes())
-                .is_err()
-            {
+            // Nonce::eq is already implemented with a constant time comparison
+            if !claims_nonce.eq(self) {
```
Comment (ramosbugs) [inline src/verification.rs:483]: does `claims_nonce != &self` (or similar) work here?

File: src/types.rs [hunk 2]
```diff
@@ -919,8 +919,12 @@ new_secret_type![
 ];
 impl PartialEq for Nonce {
     fn eq(&self, other: &Self) -> bool {
-        constant_time::verify_slices_are_equal(self.secret().as_bytes(), other.secret().as_bytes())
-            .is_ok()
+        use subtle::ConstantTimeEq;
+        self.secret()
+            .as_bytes()
+            .ct_eq(other.secret().as_bytes())
+            .unwrap_u8()
```
Comment (ramosbugs) [inline src/types.rs:926]: let's use `.into::<bool>()` or similar here as indicated in `subtle`'s [docs](https://docs.rs/subtle/2.4.1/subtle/struct.Choice.html#note)

**Review by ramosbugs — COMMENTED** [review]
thanks for splitting this out into a smaller PR!

**Comment by ramosbugs** [comment]
> # [Codecov](https://codecov.io/gh/ramosbugs/openidconnect-rs/pull/89?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=David+Ramos) Report
> > Merging [#89](https://codecov.io/gh/ramosbugs/openidconnect-rs/pull/89?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=David+Ramos) ([d1ce1f3](https://github.com/ramosbugs/openidconnect-rs/commit/d1ce1f34ee2440464fe0383b9f6bdb49949ad8dd)) into [main](https://codecov.io/gh/ramosbugs/openidconnect-rs/commit/2f79ac2ec699eff2f17a2b5c50290782623d8a9f?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=David+Ramos) ([2f79ac2](https://github.com/ramosbugs/openidconnect-rs/commit/2f79ac2ec699eff2f17a2b5c50290782623d8a9f)) will **decrease** coverage by `22.74%`.
> > The diff coverage is `100.00%`.

...that's definitely not right, Codecov.

**Comment by ramosbugs** [comment]
Hopefully #90 will help with the coverage weirdness

File: src/verification.rs (hunk 1 above)
Comment (ctron) [reply src/verification.rs:483]: True, I guess that would be the same.

I guess I wrote too much Java code lately :D

**Comment by ctron** [comment]
I updated the PR.

**Comment by ctron** [comment]
Is there something I need to do?

**Comment by ramosbugs** [comment]
Looks like CI is failing due to a clippy lint. Once it passes I'll merge the PR.

**Comment by ctron** [comment]
Sorry, fixed that. And rebased too.

**Comment by ramosbugs** [comment]
thanks!

**Pull request merged**